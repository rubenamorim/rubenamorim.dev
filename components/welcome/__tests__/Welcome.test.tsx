import { render, screen } from '@testing-library/react';

import Welcome from '../Welcome';

describe('Welcome', () => {
    it('should render correctly', () => {
        const { container } = render(<Welcome />);

        expect(screen.getByText(/Hey/)).toBeInTheDocument();
        expect(screen.getByText(/Iâ€™m Ruben,/)).toBeInTheDocument();
        expect(screen.getByText(/Software Engineer/)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
