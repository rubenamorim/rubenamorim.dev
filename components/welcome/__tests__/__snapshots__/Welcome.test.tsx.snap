// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should render correctly 1`] = `
<div>
  <div>
    <h1
      class="text-center md:text-left"
    >
      Hey
       
      <span
        aria-label="Hello"
        class="inline-block origin-bottom-right animate-wave md:animate-none md:hover:animate-wave md:focus:animate-wave"
        role="img"
      >
        👋🏻
      </span>
       
      I’m Ruben,
      <br />
      Software Engineer
    </h1>
    <p
      class="text-justify mt-12"
    >
      From
       
      <a
        class="hover:text-porto focus:text-porto transition-colors"
        href="https://visitporto.travel/"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        Porto, Portugal
      </a>
       
      to the world, I’m passionate about building things for the web. I love my profession, which I always wanted to be since I was a child and as a child I constantly try to learn new things and improve myself every day. Currently, I’m working for
       
      <a
        class="hover:text-yellow focus:text-yellow transition-colors"
        href="https://mindera.com/"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        Mindera
         
        <span
          aria-label="Mindera's yellow heart"
          role="img"
        >
          💛
        </span>
      </a>
    </p>
  </div>
</div>
`;
